### PR TITLE
GEODE-10291: Add ubuntu 22.04 (jammy) to CI

### PR DESCRIPTION
**Pull request by moleske**
I don't have ami credentials nor concourse credentials to test any of this, so I will need some help if this should be wanted in CI

### CODE REVIEW

File: docker/ubuntu-22.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        bellsoft-java11 \
+        git \
+        doxygen \
+        graphviz \
+        python3-pip \
+        clang-format-6.0 && \
+    apt-get -y autoremove && \
+    apt-get autoclean
+
+RUN pip3 install --upgrade pip && \
```
Comment (pivotal-jbarrett) [inline docker/ubuntu-22.04/Dockerfile:54]: I think you can cut out the coveralls bits. We haven't really been using it. Then you can cut out pip above too.

File: docker/ubuntu-22.04/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        libc++-dev \
+        libc++abi-dev \
+        zlib1g-dev \
+        libssl-dev \
+        bellsoft-java11 \
+        git \
+        doxygen \
+        graphviz \
+        python3-pip \
+        clang-format-6.0 && \
```
Comment (pivotal-jbarrett) [inline docker/ubuntu-22.04/Dockerfile:50]: Gosh, this should probably be clang-format-11.

**Review by pivotal-jbarrett — CHANGES_REQUESTED** [review]

File: docker/ubuntu-22.04/Dockerfile (hunk 2 above)
Comment (moleske) [reply docker/ubuntu-22.04/Dockerfile:50]: lol.  I just copied pasted the ubuntu-20.04 dockerfile, so I didn't actually look at anything beyond the from line.  I'll clean these up.

Also I don't think these are used in ci after another look, they are only here for convenience.  Not sure if anyone cares, just an observation

Comment (pivotal-jbarrett) [reply docker/ubuntu-22.04/Dockerfile:50]: Not used by CI. I occasionally use them locally to compile under the alternative platforms when trying to debug a platform specific issue. They get very little love.

Comment (moleske) [reply docker/ubuntu-22.04/Dockerfile]: per some outside of here conversation with @pdxcodemonkey it is clang-format-12.  There's a lot of inconsistency in the other dockerfiles, and clang-format-12 is not in the default repository for ubuntu 16.04 and 18.04, so will only fix it for ubuntu 22.04 for now and pr the other dockerfiles later

File: docker/ubuntu-22.04/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+RUN installer=$(mktemp) \
+    && curl -o ${installer} -L $(curl -s https://api.github.com/repos/Kitware/CMake/releases \
+        | grep -P -i 'browser_download_url.*cmake-\d+\.\d+\.\d+-linux-x86_64\.sh' \
+        | head -n 1 \
+        | cut -d : -f 2,3 \
+        | tr -d '"') \
+    && bash ${installer} --skip-license --prefix=/usr/local \
+    && rm ${installer}
+
+ARG GEODE_VERSION=1.14.4
```
Comment (gaussianrecurrence) [inline docker/ubuntu-22.04/Dockerfile:63]: Considering latest version is 1.15.0, maybe this needs to be updated

**Review by gaussianrecurrence — CHANGES_REQUESTED** [review]
Seems good, except for the version thing

File: docker/ubuntu-22.04/Dockerfile (hunk 3 above)
Comment (moleske) [reply docker/ubuntu-22.04/Dockerfile:63]: ah joys of forgetting about prs for months on end, can bump it

**Review by gaussianrecurrence — APPROVED** [review]

**Comment by gaussianrecurrence** [comment]
Hi @moleske & @pivotal-jbarrett, it seems to me that @moleke addressed all the comments, so I am not sure is this can be merged already?

**Pull request closed without merging**